src/libstd/thread.rs

```diff
@@ -416,6 +416,8 @@ impl<T: Send> JoinGuard<T> {
     }
 
     /// Detaches the child thread, allowing it to outlive its parent.
+    /// Note, however, that the main thread is special in that if it quits,
+    /// all other threads are also going to quit.
```

Comment:
I think it's misleading to say that the other threads quit, but rather just that the program terminates.